sdk/simulator-service/src/implementation/simulator-engine/reducer/swapReducer.ts

```diff
@@ -1,16 +1,88 @@
+import { TokenAmount, Price, Percentage } from '@summerfi/sdk-common/common'
 import { steps } from '@summerfi/sdk-common/simulation'
-import { addBalance, subtractBalance } from '../../helpers'
-import { SimulationState } from '../../../interfaces/simulation'
+import { addBalance, subtractBalance } from '../../utils'
+import { ISimulationState } from '../../../interfaces/simulation'
 
-export function swapReducer(step: steps.SwapStep, state: SimulationState): SimulationState {
+export function swapReducer(step: steps.SwapStep, state: ISimulationState): ISimulationState {
   const balanceWithoutFromToken = subtractBalance(step.inputs.fromTokenAmount, state.balances)
   const balanceWithToToken = addBalance(step.outputs.receivedAmount, balanceWithoutFromToken)
+
+  const baseToken = step.inputs.toTokenAmount.token
+  const quoteToken = step.inputs.fromTokenAmount.token
+
+  const offerPrice = Price.createFrom({
+    value: step.inputs.toTokenAmount
+      .toBaseUnitAsBn()
+      .div(step.inputs.fromTokenAmount.toBaseUnitAsBn())
+      .toString(),
+    baseToken,
+    quoteToken,
+  })
+
+  const spotPriceOfToToken = step.inputs.prices.find((price) =>
+    price.baseToken.address.equals(baseToken.address),
+  )
+  const spotPriceOfFromToken = step.inputs.prices.find((price) =>
+    price.baseToken.address.equals(quoteToken.address),
+  )
+
+  if (!spotPriceOfToToken || !spotPriceOfFromToken) {
+    throw new Error('Spot price for either From/To token could not be found')
+  }
+
+  const marketPrice = Price.createFrom({
+    value: spotPriceOfToToken.toBN().div(spotPriceOfFromToken.toBN()).toString(),
+    baseToken: step.inputs.fromTokenAmount.token,
+    quoteToken: step.inputs.toTokenAmount.token,
+  })
+
   return {
     ...state,
     steps: {
       ...state.steps,
       [step.name]: step,
     },
+    swaps: {
+      ...state.swaps,
+      [step.name]: {
+        provider: step.inputs.provider,
+        // Note: Can add routes back in later if we need them for the UI
+        // routes: step.inputs.routes,
+        fromTokenAmount: step.inputs.fromTokenAmount,
+        toTokenAmount: step.inputs.toTokenAmount,
+        slippage: Percentage.createFrom({ value: step.inputs.slippage.value }),
+        offerPrice,
+        marketPrice,
+        priceImpact: calculatePriceImpact(marketPrice, offerPrice),
+        summerFee: TokenAmount.createFrom({
+          token: step.inputs.fromTokenAmount.token,
+          amount: step.inputs.fromTokenAmount.multiply(step.inputs.fee.value).amount,
```

Comment:
Just as a note for future refactor: we should either accept a Percentage for multiply, or have a Utils module that allows us to do this